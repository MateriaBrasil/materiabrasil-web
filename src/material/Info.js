import React from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'

export default ({ current }) => {
  const { name, imageUrl, description, averagePrice, code } = current

  return (
    <Grid container spacing={24} justify="center">
      <Grid item xs={12} style={{ textAlign: 'center' }}>
        <Typography variant="display2">{name}</Typography>
      </Grid>
      <Grid item xs={12} sm={6} lg={8} style={{ textAlign: 'center' }}>
        <img src={imageUrl} alt={name} style={{ width: '100%' }} />
      </Grid>
      <Grid item xs={12} sm={6} lg={4}>
        <Typography variant="caption" style={{ marginBottom: 15 }}>
          {code}
        </Typography>
        <Typography variant="subheading" style={{ marginBottom: 15 }}>
          {description}
        </Typography>
        <Typography variant="subheading">{averagePrice}</Typography>
      </Grid>
    </Grid>
  )
}
