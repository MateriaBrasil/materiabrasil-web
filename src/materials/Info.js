import React from 'react'
import Typography from '@material-ui/core/Typography'

const Info = ({ style, ...props }) => (
  <div style={{ ...style, padding: 20, textAlign: 'center' }}>
    <Typography variant="headline" {...props} />
  </div>
)

export default Info
