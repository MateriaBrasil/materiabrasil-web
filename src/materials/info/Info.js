import React, { Component, Fragment } from 'react'
import find from 'lodash/find'
import Grid from '@material-ui/core/Grid'
import { Helmet } from 'react-helmet'

import Comments from '../../comments/Comments'

import checkReloadAndFetch from '../../checkReloadAndFetch'
import CoverImage from './CoverImage'
import Description from './Description'
import Sidebar from './Sidebar'
import Images from './Images'

export default class extends Component {
  componentDidUpdate(prevProps) {
    checkReloadAndFetch(this.props)
  }

  render() {
    const { current, currentUser } = this.props
    const { id, supplierId, name, listImageUrl } = current
    const { suppliers } = currentUser || {}
    const editable = find(suppliers, supplier => supplier.id === supplierId)
    document.documentElement.scrollTop = 0

    return (
      <Fragment>
        <Helmet>
          <title>{name}</title>
          <meta property="og:image" content={listImageUrl} />
        </Helmet>
        <Grid container spacing={16}>
          <CoverImage {...current} editable={editable} />
          {editable && <Images {...this.props} {...current} />}
          <Grid item xs={12} lg={7} style={{ marginBottom: 16 }}>
            <Description {...this.props} {...current} editable={editable} />
            <Comments id={id} type="materials" {...this.props} />
          </Grid>
          <Sidebar {...this.props} {...current} />
        </Grid>
      </Fragment>
    )
  }
}
