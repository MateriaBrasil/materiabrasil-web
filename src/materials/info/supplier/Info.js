import React from 'react'
import Grid from '@material-ui/core/Grid'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'

import SignUpLink from '../../../auth/SignUpLink'
import CaptionWithText from '../CaptionWithText'

export default props => {
  const { manufacturingLocation, averagePrice, supplier } = props
  const { currentUser, location } = props
  const { name, email } = supplier || {}
  const { pathname } = location

  return (
    <Grid item xs={12} lg={3} style={{ marginBottom: 16 }}>
      <Card>
        <CardContent>
          <CaptionWithText caption="Fornecedor" text={name} />
          <CaptionWithText
            caption="Local de produção"
            text={manufacturingLocation}
          />
          <CaptionWithText caption="Preço médio" text={averagePrice} />
          {currentUser ? (
            <CaptionWithText caption="Contato do fornecedor" text={email} />
          ) : (
            <SignUpLink
              pathname={pathname}
              text="para ver o contato do fornecedor."
            />
          )}
        </CardContent>
      </Card>
    </Grid>
  )
}
