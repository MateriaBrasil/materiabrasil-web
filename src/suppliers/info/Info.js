import React, { Component } from 'react'
import Link from 'react-router-dom/Link'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'

import checkReloadAndFetch from '../../checkReloadAndFetch'
import Avatar from '../../imageUpload/Avatar'
import Addresses from '../../addresses/Addresses'

import InfoCard from './InfoCard'
import Materials from './Materials'

class Info extends Component {
  componentDidUpdate(prevProps) {
    checkReloadAndFetch(this.props)
  }
  render() {
    const { current, currentUser } = this.props
    const { userId, materials, id, name, imageUrl } = current
    const editPath =
      currentUser && userId === currentUser.id && `/suppliers/${id}/avatar`
    const editProfile =
      currentUser && currentUser.id.toString() === userId.toString()

    return (
      <Grid container spacing={32}>
        <Grid item xs={12} lg={4}>
          <Avatar name={name} editPath={editPath} imageUrl={imageUrl} />
          <Typography
            variant="display1"
            style={{ marginTop: 16, marginBottom: 16 }}
          >
            {name}
          </Typography>
          {editProfile ? (
            <Link to={`/suppliers/${id}/edit`}>
              <Typography variant="subheading">Editar perfil</Typography>
            </Link>
          ) : null}
        </Grid>
        <Grid item xs={12} lg={8}>
          <InfoCard {...this.props} />
        </Grid>
        <Grid item xs={12}>
          {materials && <Materials list={materials} />}
        </Grid>
        <Addresses id={id} supplier={current} {...this.props} />
      </Grid>
    )
  }
}

export default Info
