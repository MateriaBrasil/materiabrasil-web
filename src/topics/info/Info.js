import React, { Fragment } from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

import moment from '../../moment'
import Comments from '../../comments/Comments'

export default props => {
  const { id, user, subject, text, createdAt } = props
  const { firstName, lastName } = user
  const name = `${firstName} ${lastName}`

  return (
    <Fragment>
      <Card>
        <CardContent>
          <Typography variant="display1" style={{ marginBottom: 8 }}>
            {subject}
          </Typography>
          <Typography variant="caption" style={{ marginBottom: 24 }}>
            {`Criado por ${name} hรก ${moment(createdAt).fromNow()}`}
          </Typography>
          <Typography variant="body1">{text}</Typography>
        </CardContent>
      </Card>
      <Typography
        variant="headline"
        style={{ marginTop: 24, marginBottom: 24 }}
      >
        Respostas
      </Typography>
      <Comments id={id} type="topics" term="resposta" {...props} />
    </Fragment>
  )
}
