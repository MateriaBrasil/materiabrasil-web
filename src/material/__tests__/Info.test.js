import React from 'react'
import renderer from 'react-test-renderer'

import Info from '../Info'

it('renders correctly', () => {
  const current = {
    name: 'foo',
    imageUrl: 'bar',
    description: 'info',
    averagePrice: '100',
    code: '999',
  }
  const tree = renderer.create(<Info current={current} />).toJSON()
  expect(tree).toMatchSnapshot()
})
