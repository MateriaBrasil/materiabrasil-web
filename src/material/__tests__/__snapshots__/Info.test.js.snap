// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-24 MuiGrid-justify-xs-center-18"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-39"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h1
      className="MuiTypography-root-92 MuiTypography-display2-95"
    >
      foo
    </h1>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-39 MuiGrid-grid-sm-6-46 MuiGrid-grid-lg-8-74"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <img
      alt="foo"
      src="bar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-39 MuiGrid-grid-sm-6-46 MuiGrid-grid-lg-4-70"
  >
    <span
      className="MuiTypography-root-92 MuiTypography-caption-102"
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      999
    </span>
    <h3
      className="MuiTypography-root-92 MuiTypography-subheading-99"
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      info
    </h3>
    <h3
      className="MuiTypography-root-92 MuiTypography-subheading-99"
    >
      100
    </h3>
  </div>
</div>
`;
